### PR TITLE
Feature/id 286/report traffic

### PR DESCRIPTION
**Pull request by GastonGutierrezC**
What did I do?

I made the functionality that the user just by pressing the traffic button, automatically saves the traffic direction, estimates the time and tells the level of traffic, automatically.

Why did I do it?

So that the user does not have to manually set the two traffic points, nor does he have to estimate the time or tell the type of traffic.

How did I do it?

Adding action to the report traffic button, creating classes for the automatic saving of the data and creating test cases for it.

### CODE REVIEW

File: app/src/main/java/com/isc/hermes/controller/MapWayPointController.java [hunk 1]
```diff
@@ -21,6 +34,7 @@ public class MapWayPointController implements MapClickConfigurationController {
     private final WaypointOptionsController waypointOptionsController;
     private boolean isMarked;
     private Context context;
+    private int markerCount = 0;
```
Comment (JhaelArce) [inline app/src/main/java/com/isc/hermes/controller/MapWayPointController.java]: markerCount should be initialized in the constructor

File: app/src/main/java/com/isc/hermes/controller/MapWayPointController.java [hunk 2]
```diff
@@ -49,23 +63,42 @@ public boolean onMapClick(@NonNull LatLng point) {
...
     }
 
     /**
      * Method to perform action to click on map
      * @param point Is point passed as parameter with its latitude and longitude
      */
-    private void doMarkOnMapAction(LatLng point){
-        if (isMarked){
+
+    private void doMarkOnMapAction(LatLng point) {
```
Comment (JhaelArce) [inline app/src/main/java/com/isc/hermes/controller/MapWayPointController.java:77]: You should refactor this method, there are parts of code that are repeated and can go to the other method

**Review by jhael-arce-jalafund — APPROVED** [review]

File: app/src/main/java/com/isc/hermes/controller/MapWayPointController.java (hunk 1 above)
Comment (GastonGutierrezC) [reply app/src/main/java/com/isc/hermes/controller/MapWayPointController.java]: Done

File: app/src/main/java/com/isc/hermes/controller/MapWayPointController.java (hunk 2 above)
Comment (GastonGutierrezC) [reply app/src/main/java/com/isc/hermes/controller/MapWayPointController.java:77]: Done

**Review by EmanuelJGalindo — APPROVED** [review]
Looks like there's no observations for me.

**Review by saletG08 — APPROVED** [review]
I don't have any comments

**Pull request merged**